### PR TITLE
Changes to color generation of graphs

### PR DESCRIPTION
**Pull request by Centauri0320001**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Ensured Constellation's menus and menu items have a suitable colour scheme for colourblind users. From testing the only two problematic areas revolved around graphs and node generation. Hence, an additional menu for a colorblind mode selection in Setup -> Options -> Constellation -> Application was added. Switching to a colorblind scheme now has several effects for the application: 

- The way random colors are generated for nodes in draw mode has been updated to be more colorblind friendly, and prevents colors which are too dark from being generated for contrast against the black background. Generation of certain shades has also been prevented based on the colorblind selection to increase contrast for colorblind users. 

- Changes the default colors of specific node/transaction types when a new graph is loaded to account for specific colorblind types. 

- Users can update their graphs to currently selected colourblind scheme using Display -> Apply colourblind schema to current graph (See alternate designs)

image

Updated color scheme to improve the contrast for users with colorblindness. For example the new deuteranopic color scheme for predefined vertex types as how most users will see it:
image
Deuteranopic users will see it as: (This is a simulated colorblindness filter)
image

### Alternate Designs

Automatic dynamic colour swapping of schema types was considered. However, the constraint of the current way schema types are set upon compilation time means the colour attributes of the nodes has to be explicitly modified to apply the Colourblind schemas without restarting Constellation. 

### Why Should This Be In Core?

Adds options to enhance usability of the application for users with colorblindness.
### Benefits

Usability of Constellation will improve for users with colorblindness as the contrast of graph elements will improve.
### Possible Drawbacks

- Colorblind users may find the new color schemes to be a bit jarring if they are accustomed to the current default colors of certain node and transaction types. 

### Verification Process

Oracle color was used as a colorblindness simulator to test the changes. Additionally, a colleague with colorblindness was able to verify that the color scheme works for their type.
### Applicable Issues

#46

### CODE REVIEW

File: CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java [hunk 1]
```diff
@@ -50,15 +53,18 @@ public final class ConstellationColor implements Comparable<ConstellationColor>,
     public static final ConstellationColor DARK_GREEN = new ConstellationColor("DarkGreen", 0.0F, 0.5F, 0.0F, 1.0F);
     public static final ConstellationColor DARK_GREY = new ConstellationColor("DarkGrey", Color.DARK_GRAY);
     public static final ConstellationColor DARK_ORANGE = new ConstellationColor("DarkOrange", 1.0F, 0.5F, 0.25F, 1.0F);
+    public static final ConstellationColor DARK_PURPLE = new ConstellationColor("Darkpurple", 51, 26, 128, 255);
```
Comment (Delphinus8821) [inline CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java:56]: Change to DarkPurple

File: CoreVisualSchema/src/au/gov/asd/tac/constellation/graph/schema/visual/VisualSchemaFactory.java [hunk 2]
```diff
@@ -297,7 +300,44 @@ public void completeTransaction(final GraphWriteMethods graph, final int transac
...
+                        randFloat3 += brightenFloat;
+                    }
+
+                    return ConstellationColor.getColorValue(randFloat1, 0F, randFloat3, 1.0F);
+                case "Protanopia":
+                    if (randFloat1 <= lowFloat && randFloat3 <= lowFloat) {
+                        randFloat3 += brightenFloat;
+                        randFloat1 += brightenFloat;
+                    }
+                    return ConstellationColor.getColorValue(randFloat1, 0F, randFloat3, 1.0F);
```
Comment (OrionsGuardian) [inline CoreVisualSchema/src/au/gov/asd/tac/constellation/graph/schema/visual/VisualSchemaFactory.java:330]: Both these cases are essentially running the same code ... is this correct ?
If so, the code can be simplified by stripping out lines 318 to 324,
so the two cases go into the same code block.

**Review by OrionsGuardian — CHANGES_REQUESTED** [review]

**Review by antares1470 — CHANGES_REQUESTED** [review]
A few things:
1) Are you able to add an entry to `whatsnew.txt` for this one?
2) I'm noting that some of the transaction type colors are retaining the color that they had in non-colorblind mode whereas they are being changed when used in the node types (e.g. the emerald color in the communication transaction type remains the same but is changed to buttermilk for the telephone identifier vertex type). Not sure if this was intentional (noting similar things for azure, musk, and banana). Similarly, you have one vertex type changing TO brown, and two transaction types changing FROM brown
3) Somewhat related to the above, it seems colors change to different colors in different situations (e.g. carrot becomes blue for location nodes, but becomes navy for location transactions). Not necessarily a problem, but just an observation
4) Noting that at present, there doesn't appear to be any difference in how Constellation handles Deuteranopia and Protanopia. In that case, might be worth combining the 2 into one option

**Review by Delphinus8821 — APPROVED** [review]

**Review by antares1470 — CHANGES_REQUESTED** [review]
Code wise it seems ok but having done some testing, it looks like colors on existing graph elements on existing graphs don't get updated. This usually requires adding a schema updater (which will still be needed for older graphs if you're able to write one), but even after creating new graphs this issue still arises.
e.g.
1) With color blind mode applied, create a graph with one node being telephone identifier type. Save the graph. The color of the node should be emerald
2) Open preferences and change to one of the color blind modes (e.g. Deuteranopia)
3) Restart Constellation
4) Open your saved graph. You would expect the node to now be buttermilk color but it is still emerald, even after a complete with schema. Manually change the again to Telephone identifier and then after a complete with schema it will now be buttermilk

Basically it seems that rather than color being dynamically determined all the time depending on the color blind mode set, it is instead statically set with the color simply determined by the color blind mode at the time of graph element creation. If it possible to make this color swap more dynamic, I think that would be nice addition

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+@ActionRegistration(displayName = "#CTL_ApplyColorblindAction", iconBase = "au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/resources/colorblind.png", surviveFocusChange = true)
+@ActionReference(path = "Menu/Display", position = 950, separatorBefore = 901)
+@NbBundle.Messages("CTL_ApplyColorblindAction=Apply Selected Colorblind Schema to Graph")
+public final class ApplyColorblindAction extends SimpleAction {
+
+    private static Preferences prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+    //Retrieve colorblind mode selection preference 
+    private static String colorMode = prefs.get(ApplicationPreferenceKeys.COLORBLIND_MODE, ApplicationPreferenceKeys.COLORBLIND_MODE_DEFAULT);
+
+    public ApplyColorblindAction(GraphNode context) {
```
Comment (Delphinus8821) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java:58]: final

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        super(context);
+    }
+
+    @Override
+    protected void edit(PluginGraphs graphs, PluginInteraction interaction, PluginParameters parameters) throws InterruptedException, PluginException {
```
Comment (Delphinus8821) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java:63]: finals

**Review by Delphinus8821 — CHANGES_REQUESTED** [review]

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+ */
+@ActionID(category = "Display", id = "au.gov.asd.tac.constellation.graph.visual.plugins.colorblind.ApplyColorblindAction")
+@ActionRegistration(displayName = "#CTL_ApplyColorblindAction", iconBase = "au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/resources/colorblind.png", surviveFocusChange = true)
+@ActionReference(path = "Menu/Display", position = 950, separatorBefore = 901)
+@NbBundle.Messages("CTL_ApplyColorblindAction=Apply Selected Colorblind Schema to Graph")
+public final class ApplyColorblindAction extends SimpleAction {
+
+    private static Preferences prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+    //Retrieve colorblind mode selection preference 
+    private static String colorMode = prefs.get(ApplicationPreferenceKeys.COLORBLIND_MODE, ApplicationPreferenceKeys.COLORBLIND_MODE_DEFAULT);
```
Comment (antares1470) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java:56]: I think these 2 variables can just be defined in the `run` function since they aren't used outside of that function (they can also be made final in that instance)

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+            final int vertexCount = graph.getVertexCount();
+            for (int vertex = 0; vertex < vertexCount; vertex++) {
+
+                final int vxId = graph.getVertex(vertex);
+                final SchemaVertexType vertexType = graph.getObjectValue(typeAttributeId, vxId);
+
+                if (typeAttributeId != Graph.NOT_FOUND) {
+                    ConstellationColor caseColor;
+
+                    switch (vertexType.getName()) {
```
Comment (antares1470) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java:92]: is there a way to be able to write this section in a more generic sense? Noting that Constellation is extendable, people may want to be able to have this function apply for their own custom types beyond just what is available in Core (i.e. this function should work for any schema defined on top of Constellation)

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+        final int transactionColorAttr = VisualConcept.TransactionAttribute.COLOR.get(graph);
+
+        if (transactionTypeId != Graph.NOT_FOUND) {
+            final int transactionCount = graph.getTransactionCount();
+            for (int transaction = 0; transaction < transactionCount; transaction++) {
+
+                final int transactionId = graph.getTransaction(transaction);
+                final SchemaTransactionType transactionType = graph.getObjectValue(transactionTypeId, transactionId);
+
+                if (transactionTypeId != Graph.NOT_FOUND) {
```
Comment (antares1470) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java:149]: similar case here

File: CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/whatsnew.txt [hunk 8]
```diff
@@ -1,9 +1,15 @@
 == 3030-12-31 Getting Started
 <p>If you're new to Constellation, read the <a href="" helpId="au.gov.asd.tac.constellation.functionality.gettingstarted">getting started guide</a>.</p>
 
+== 2023-09-28 Hide empty attributes in the Attribute Editor View
+<p>The Attribute Editor view now allows for empty (null) attributes to be hidden to de-clutter the view.</p>
+<p>The heading of each attribute category has replaced the "Show all" toggle with a "Show Hidden" toggle and a new "Show Empty" toggle has been added. The "Show Empty" toggle is enabled each time Constellation is restarted to ensure users make a conscious decision to hide them.</p>
+<p>Attributes that are marked as hidden will only be shown if the "Show Hidden" toggle is enabled while attributes that have no value set (and are empty) will only be shown if the "Show Empty" toggle is enabled. Hidden and empty attributes require both toggles to be enabled.</p>
+
 == 2023-09-15 Colorblind Mode in Options
```
Comment (antares1470) [inline CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/whatsnew.txt:9]: may be worth changing the date on this one (and moving it up)

**Review by antares1470 — CHANGES_REQUESTED** [review]
Not sure what could be done here, but I note that the action is the only way for old graphs to get the new colorblind settings applied. Not sure if there is a way to have that automatically applied when you open the graph (as in automatically applying whatever the current colorblind mode is)

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+        final int txColorRef = VisualConcept.GraphAttribute.TRANSACTION_COLOR_REFERENCE.ensure(wg);
+
+        wg.setStringValue(vxColorRef, 0, vxColorAttrName);
+        wg.setStringValue(txColorRef, 0, txColorAttrName);
+    }
+
+    /*Adjust RGB values using the to-be removed RGB value as a proportion of the calculation, acting as contrast booster for brightness adjustments.  
+        Evaluate the selected colorblind mode and adjust contrast if RGB value is high enough; prevents new color from being too dark, then remove imperceivable colors. 
+        Primary colors for the modes are then adjusted at different strengths to improve contrast. I.E. remove 50% red in deut, remove 18% blue for prot.*/
+    public static ConstellationColor calcColorBrightness(ConstellationColor vertexColor) {
```
Comment (Delphinus8821) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java:86]: parameter final

**Review by Delphinus8821 — CHANGES_REQUESTED** [review]
Looks like it's working well for drawing new nodes, and updating graphs with only nodes on it. Got an error when testing a graph with lots of transactions 
Edit: this error only occurs if a graph is created where a single transaction has been given multiple types. This was tested using a complete graph builder. Not sure if there are any other instances where transactions can be created with multiple types so we may be able to ignore this. 
image

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+        final int txColorRef = VisualConcept.GraphAttribute.TRANSACTION_COLOR_REFERENCE.ensure(wg);
+
+        wg.setStringValue(vxColorRef, 0, vxColorAttrName);
+        wg.setStringValue(txColorRef, 0, txColorAttrName);
+    }
+
+    /*Adjust RGB values using the to-be removed RGB value as a proportion of the calculation, acting as contrast booster for brightness adjustments.  
+        Evaluate the selected colorblind mode and adjust contrast if RGB value is high enough; prevents new color from being too dark, then remove imperceivable colors. 
+        Primary colors for the modes are then adjusted at different strengths to improve contrast. I.E. remove 50% red in deut, remove 18% blue for prot.*/
+    public static final ConstellationColor calcColorBrightness(ConstellationColor vertexColor) {
```
Comment (Delphinus8821) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java:86]: final

File: CorePreferences/release/modules/ext/docs/CorePreferences/src/au/gov/asd/tac/constellation/preferences/application-preferences.md [hunk 11]
```diff
@@ -67,3 +67,11 @@ directory.
 
 This is where the user is able to set their preferred default font and
 default font size for the Constellation application.
+
+## Colorblind Mode
+
+The user is able to set a colorblind mode for constellation. When a 
+colorblind mode is set, draw nodes are automatically generated with
```
Comment (Delphinus8821) [inline CorePreferences/release/modules/ext/docs/CorePreferences/src/au/gov/asd/tac/constellation/preferences/application-preferences.md:74]: nodes drawn

**Review by Delphinus8821 — CHANGES_REQUESTED** [review]

**Review by Delphinus8821 — APPROVED** [review]

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java [hunk 12]
```diff
@@ -15,100 +15,60 @@
...
+    private ColorblindUtilities() {
+        throw new IllegalStateException("Utility class");
     }
 
-    public void run(final WritableGraph graph) {
-        final int vxColorAttr = VisualConcept.VertexAttribute.COLOR.get(graph);
-        final int txColorAttr = VisualConcept.TransactionAttribute.COLOR.get(graph);
+    public static void colorNodes(final GraphWriteMethods wg, final int vxColorblindAttr, final int txColorblindAttr) {
+        Preferences prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+        String COLORMODE = prefs.get(ApplicationPreferenceKeys.COLORBLIND_MODE, ApplicationPreferenceKeys.COLORBLIND_MODE_DEFAULT);
```
Comment (antares1470) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java:37]: finals, also can `COLORMODE` can be made camel case

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java [hunk 13]
```diff
@@ -15,100 +15,60 @@
...
+        final int transactionCount = wg.getTransactionCount();
 
         if (!"None".equals(COLORMODE)) {
             //Iterate through graph vertices. If vertexType is a defined schemaType color will be adjusted if applicable.
             for (int vertex = 0; vertex < vertexCount; vertex++) {
-                final int vxId = graph.getVertex(vertex);
-                final ConstellationColor vertexColor = graph.getObjectValue(vxColorAttr, vxId);
-                final ConstellationColor vxColorblindAlpha = graph.getObjectValue(vxColorblindAttr, vxId);
+                final int vxId = wg.getVertex(vertex);
+                final ConstellationColor vertexColor = wg.getObjectValue(vxColorAttr, vxId);
```
Comment (antares1470) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java:49]: I would either add a check for `vxColorAttr != Graph.NOT_FOUND` or use ensure instead of get

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java [hunk 14]
```diff
@@ -15,100 +15,60 @@
...
                 }
             }
-
+            //Iterate through graph transactions. If transcationType is a defined schemaType color will be adjusted if applicable.
             for (int transaction = 0; transaction < transactionCount; transaction++) {
-                final int transactionId = graph.getTransaction(transaction);
-                final ConstellationColor transactionColor = graph.getObjectValue(txColorAttr, transactionId);
-                final ConstellationColor txColorblindAlpha = graph.getObjectValue(vxColorblindAttr, transactionId);
+                final int transactionId = wg.getTransaction(transaction);
+                final ConstellationColor transactionColor = wg.getObjectValue(txColorAttr, transactionId);
```
Comment (antares1470) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java:60]: same case here

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java [hunk 15]
```diff
@@ -15,100 +15,60 @@
...
 
         if (!"None".equals(COLORMODE)) {
             //Iterate through graph vertices. If vertexType is a defined schemaType color will be adjusted if applicable.
             for (int vertex = 0; vertex < vertexCount; vertex++) {
-                final int vxId = graph.getVertex(vertex);
-                final ConstellationColor vertexColor = graph.getObjectValue(vxColorAttr, vxId);
-                final ConstellationColor vxColorblindAlpha = graph.getObjectValue(vxColorblindAttr, vxId);
+                final int vxId = wg.getVertex(vertex);
+                final ConstellationColor vertexColor = wg.getObjectValue(vxColorAttr, vxId);
+                final ConstellationColor vxColorblindAlpha = wg.getObjectValue(vxColorblindAttr, vxId);
```
Comment (antares1470) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java:50]: noting that the attribute here comes via parameter, probably be good to check if the attribute exists

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java [hunk 16]
```diff
@@ -15,100 +15,60 @@
...
             }
-
+            //Iterate through graph transactions. If transcationType is a defined schemaType color will be adjusted if applicable.
             for (int transaction = 0; transaction < transactionCount; transaction++) {
-                final int transactionId = graph.getTransaction(transaction);
-                final ConstellationColor transactionColor = graph.getObjectValue(txColorAttr, transactionId);
-                final ConstellationColor txColorblindAlpha = graph.getObjectValue(vxColorblindAttr, transactionId);
+                final int transactionId = wg.getTransaction(transaction);
+                final ConstellationColor transactionColor = wg.getObjectValue(txColorAttr, transactionId);
+                final ConstellationColor txColorblindAlpha = wg.getObjectValue(txColorblindAttr, transactionId);
```
Comment (antares1470) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java:61]: same here

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java [hunk 17]
```diff
@@ -123,7 +83,10 @@ public static void setColorRef(final GraphWriteMethods wg, final String vxColorA
     /*Adjust RGB values using the to-be removed RGB value as a proportion of the calculation, acting as contrast booster for brightness adjustments.  
         Evaluate the selected colorblind mode and adjust contrast if RGB value is high enough; prevents new color from being too dark, then remove imperceivable colors. 
         Primary colors for the modes are then adjusted at different strengths to improve contrast. I.E. remove 50% red in deut, remove 18% blue for prot.*/
-    public ConstellationColor calcColorBrightness(ConstellationColor vertexColor) {
+    public static final ConstellationColor calcColorBrightness(final ConstellationColor vertexColor) {
+        Preferences prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+        String COLORMODE = prefs.get(ApplicationPreferenceKeys.COLORBLIND_MODE, ApplicationPreferenceKeys.COLORBLIND_MODE_DEFAULT);
```
Comment (antares1470) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java:88]: same case as first comment

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Review by antares1470 — COMMENTED** [review]
One thing I did note from testing was that adding new nodes seems to make them all grey. I know that is the colour of the unknown node by default but was that an intentional move?

EDIT: Just having tested further it only seems to stop creating coloured nodes after 'Apply colorblind selection to graph' is done

**Comment by Centauri0320001** [comment]
> One thing I did note from testing was that adding new nodes seems to make them all grey. I know that is the colour of the unknown node by default but was that an intentional move?
> 
> EDIT: Just having tested further it only seems to stop creating coloured nodes after 'Apply colorblind selection to graph' is done

This occurs because the an additional layer is activated on the graph after the "Apply colorblind selection to graph" button is selected, the drawn nodes still have a base layer of colour that is being generated. It is intentional, allows users to give their newly generated nodes a custom colour (which is ignored by the button once a custom color is set), otherwise they can press the button again to color the newly drawn nodes.

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java (hunk 15 above)
Comment (Centauri0320001) [reply CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java:50]: Would this check be necessary even if the attribute is being ensured in the ColorblindAction class?

**Review by antares1470 — CHANGES_REQUESTED** [review]
I don't know how possible it is to fix but I've noticed that despite having a colorblind mode selected and then restarting Constellation, its still defaulting to the non-colorblind color of the schema type. Similarly, once colorblind mode is applied, the subsequent grey nodes created also remain grey even after changing the type.

For types defined in the schema, I think it would be useful to ensure complete with schema colors the type according to what mode is set when first launched as its currently not doing that (contrary to what it says in preferences).

See Gif (it's hard to read but I've got the mode set to Protonopia here):

colorblindSchemaIssue

**Review by Delphinus8821 — APPROVED** [review]

**Review by antares1470 — COMMENTED** [review]
Really liking the inclusion of embedding into Complete with Schema.
My only question then is with doing that, do we still need the "Apply colorblind schema to graph" action? Unless you perform the action instead of completing with schema, the action otherwise doesn't do anything

**Review by antares1470 — CHANGES_REQUESTED** [review]
If you complete with schema after having None selected as the colorblind mode, all the nodes go grey (including those with types belonging to schema)

**Review by antares1470 — APPROVED** [review]
